File: packages/amplify_datastore/android/src/test/kotlin/com/amazonaws/amplify/amplify_datastore/FlutterSerializedModelData.kt
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.amazonaws.amplify.amplify_datastore

import com.amplifyframework.core.model.SerializedCustomType
import com.amplifyframework.core.model.SerializedModel

val blogSerializedModel = SerializedModel.builder()
    .modelSchema(
        blogSchema
    )
    .serializedData(
        mapOf(
            "id" to "999",
            "name" to "blog name"
        )
    ).build()

var commentSerializedModel = SerializedModel.builder()
    .modelSchema(
        commentSchema
    )
    .serializedData(
        mapOf(
            "id" to "999",
            "post" to mapOf(
                "id" to "555"
            ),
            "content" to "content"
        )
    ).build()

var postSerializedModel = SerializedModel.builder()
    .modelSchema(
        postSchema
    )
    .serializedData(
        mapOf(
            "id" to "999",
            "title" to "post title",
            "created" to "2020-11-25T01:28:49.000Z",
            "blog" to mapOf(
                "id" to "555"
            )
        )
    ).build()

var allTypeModelSerializedModel = SerializedModel.builder()
    .modelSchema(
        allTypeModelSchema
    )
    .serializedData(
        mapOf(
            "id" to "999",
            "stringType" to "string value",
            "intType" to 10,
            "floatType" to 9.99,
            "boolType" to true,
            "dateType" to "2020-09-09",
            "dateTimeType" to "2020-11-25T01:28:49.000Z",
            "timeType" to "20:20:20:020",
            "timestampType" to 999,
            "enumType" to "maybe"
        )
    ).build()

val personSerializedModel = SerializedModel.builder()
    .modelSchema(personSchema)
    .serializedData(
        mapOf(
            "id" to "123",
            "name" to "Tester Testing",
            "contact" to SerializedCustomType.builder()
                .serializedData(
                    mapOf(
                        "email" to "test@testing.com",
                        "phone" to SerializedCustomType.builder()
                            .serializedData(
                                mapOf(
                                    "country" to "+1",
                                    "area" to "415",
                                    "number" to "6666666"
                                )
                            )
                            .customTypeSchema(phoneSchema)
                            .build(),
                        "mailingAddresses" to listOf(
                            SerializedCustomType.builder()
                                .serializedData(
                                    mapOf(
                                        "line1" to "000 Somewhere far",
                                        "line2" to "apt 4",
                                        "city" to "San Francisco",
                                        "state" to "CA",
                                        "postalCode" to "94115"
                                    )
                                )
                                .customTypeSchema(addressSchema)
                                .build(),
                            SerializedCustomType.builder()
                                .serializedData(
                                    mapOf(
                                        "line1" to "111 Somewhere close",
                                        "line2" to null,
                                        "city" to "Seattle",
                                        "state" to "WA",
                                        "postalCode" to "98101"
                                    )
                                )
                                .customTypeSchema(addressSchema)
                                .build()
                        )
                    )
                )
                .customTypeSchema(contactSchema)
                .build(),
            "propertiesAddresses" to listOf(
                SerializedCustomType.builder()
                    .serializedData(
                        mapOf(
                            "line1" to "222 Somewhere in the middle",
                            "line2" to null,
                            "city" to "Portland",
                            "state" to "OR",
                            "postalCode" to "97035"
                        )
                    )
                    .customTypeSchema(addressSchema)
                    .build()
            )
        )
    )
    .build()

File: packages/amplify_datastore/example/ios/unit_tests/resources/FlutterSerializedModelData.swift
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import Amplify
@testable import AmplifyPlugins
@testable import amplify_datastore

struct FlutterSerializedModelData {
    static var BlogSerializedModel: FlutterSerializedModel =
        .init(map: [
            "id": JSONValue.string("999"),
            "name": JSONValue.string("blog name"),
        ], modelName: "Blog")
    static var CommentSerializedModel: FlutterSerializedModel =
        .init(map: [
            "id": JSONValue.string("999"),
            "post": JSONValue.object([
                "id": "555"
            ]),
            "content": "content"
       ], modelName: "Comment")

    static var PostSerializedModel: FlutterSerializedModel =
        .init(map: [
            "id": JSONValue.string("999"),
            "title": JSONValue.string("post title"),
            "created": JSONValue.string("2020-11-25T01:28:49.000Z"),
            "blog": JSONValue.object([
                "id": "555",
                "name": "Blog 1"
            ]),
            "author": JSONValue.object([
                "id": "666",
                "name": "Someone Testing",
                "department": JSONValue.object([
                    "id": "777",
                    "name": "Department 1",
                    "description": "Testing department 1"
                ])
            ])
        ], modelName: "Post")

    static var AllTypeModelSerializedModel: FlutterSerializedModel =
        .init(map: [
            "id": JSONValue.string("999"),
            "stringType": JSONValue.string("string value"),
            "intType": JSONValue.number(10),
            "floatType": JSONValue.number(9.99),
            "boolType": JSONValue.boolean(true),
            "dateType": JSONValue.string("2020-09-09Z"),
            "dateTimeType": JSONValue.string("2020-11-25T01:28:49.000Z"),
            "timeType": JSONValue.string("20:20:20:020Z"),
            "timestampType": JSONValue.number(999),
            "enumType": JSONValue.string("maybe")
        ], modelName: "AllTypeModel")

    static var PersonModelSerializedModel: FlutterSerializedModel =
        .init(
            map: [
                "id": JSONValue.string("123"),
                "name": JSONValue.string("Tester Testing"),
                "contact": JSONValue.object([
                    "email": "test@testing.com",
                    "phone": JSONValue.object([
                        "country": "+1",
                        "area": "415",
                        "number": "6666666"
                    ]),
                    "mailingAddresses": JSONValue.array([
                        JSONValue.object([
                            "line1": "000 Somewhere far",
                            "line2": "apt 4",
                            "city": "San Francisco",
                            "state": "CA",
                            "postalCode": "94115"
                        ]),
                        JSONValue.object([
                            "line1": "000 Somewhere close",
                            "line2": JSONValue.null,
                            "city": "Seattle",
                            "state": "WA",
                            "postalCode": "98101"
                        ])
                    ])
                ]),
                "propertiesAddresses": JSONValue.array([
                    JSONValue.object([
                        "line1": "222 Somewhere in the middle",
                        "line2": JSONValue.null,
                        "city": "Portland",
                        "state": "OR",
                        "postalCode": "97035"
                    ])
                ]),
                "anotherCustomTypeTree": JSONValue.object([
                    "field1": "yes another dependencies tree",
                    "field2": JSONValue.object([
                        "field": "just a string"
                    ])
                ])
            ],
            modelName: "Person"
        )
}
